mono/metadata/w32process-win32.c

```diff
@@ -53,6 +53,40 @@ mono_w32process_get_pid (gpointer handle)
 	return GetProcessId (handle);
 }
 
+#if HAVE_API_SUPPORT_WIN32_ENUM_WINDOWS
+typedef struct {
+	DWORD pid;
+	HWND hwnd;
+} EnumWindowsArgs;
+
+static BOOL CALLBACK
+mono_enum_windows_callback(HWND hwnd, LPARAM lparam)
+{
+	EnumWindowsArgs *args = (EnumWindowsArgs *)lparam;
+	DWORD pid = 0;
+	GetWindowThreadProcessId(hwnd, &pid);
+	if (pid != args->pid || GetWindow(hwnd, GW_OWNER) != NULL || !IsWindowVisible(hwnd)) return TRUE;
+	args->hwnd = hwnd;
+	return FALSE;
+}
+
+HANDLE
+ves_icall_System_Diagnostics_Process_MainWindowHandle_internal (guint32 pid, MonoError *error)
+{
+	EnumWindowsArgs args = {pid, NULL};
+	EnumWindows(mono_enum_windows_callback, (LPARAM)&args);
+	return args.hwnd;
+}
+#elif !HAVE_EXTERN_DEFINED_WIN32_ENUM_WINDOWS
+HANDLE
+ves_icall_System_Diagnostics_Process_MainWindowHandle_internal (guint32 pid, MonoError *error)
+{
+	g_unsupported_api ("EnumWindows");
+	SetLastError (ERROR_NOT_SUPPORTED);
```

Comment:
I'm not sure we have `SetLastError` support on icalls.  (ie you may not be able to see this with `GetLastWin32Error`).

If you want a `NotSupportedException`, something like this will work
```suggestion
	mono_error_set_not_supported (error, "MainWindowHandle not supported");
```


@lateralusX what do we normally do for unsupported APIs on other Windows API subsets?

Reply:
Ah yes, I missed that call when looking at how the other calls below were done.

But now I'm thinking maybe it'd be better just to return NULL here as well, to keep the current behavior until there's an implementation for it on UWP.

Reply:
GetLastError are currently used in p/invokes (and for internal error handling). The important part returning error in icall's is to use mono_error_set_not_supported, we also do the g_unsupported_api to get some logging:

g_unsupported_api ("EnumWindows");
mono_error_set_not_supported (error, "MainWindowHandle not supported");